frontend/src/components/StakeSelectorItem.vue

```diff
@@ -1,217 +1,874 @@
+        </template>
+      </multiselect>
+      <button class="stake-button stake-submit-button"
+        @click="onSubmit"
+        :disabled="currentState !== CurrentState.ok">
+        <span v-if="isLoadingStake"><i class="fa fa-spinner fa-spin"></i></span>
+        <span v-else>{{ submitButtonLabel }}</span>
+        </button>
+    </div>
   </div>
 </template>
 
-<script>
+<script lang="ts">
+import Vue from 'vue';
 import { formatDurationFromSeconds } from '../utils/date-time';
-import { isNftStakeType } from '../interfaces/State';
-
-export default {
-  props: [
-    'stakeTitle',
-    'stakeTokenName',
-    'rewardTokenName',
-    'stakeType',
-    'minimumStakeTime',
-    'estimatedYield',
-    'rewardsDuration',
-    'deprecated',
-    'rewardDistributionTimeLeft',
-    'currentRewardEarned'
-  ],
+import { isStakeType, isNftStakeType, StakeType, NftStakeType } from '../interfaces/State';
+import { toBN } from '@/utils/common';
+import { secondsToDDHHMMSS } from '../utils/date-time';
+import {Multiselect} from 'vue-multiselect';
+import {PropType} from 'vue/types/options';
+import BN from 'bignumber.js';
+import {mapActions, mapState} from 'vuex';
+import i18n from '@/i18n';
+import { TranslateResult } from 'vue-i18n';
+
+interface StoreMappedActions {
+  fetchStakeDetails(payload: {stakeType: StakeType | NftStakeType}):
+  Promise<void>;
+  stake(payload: {amount: string; stakeType: StakeType | NftStakeType}):
+  void;
+  stakeNfts(payload: {ids: string[]; stakeType: StakeType | NftStakeType}):
+  void;
+  unstake(payload: {amount: string; stakeType: StakeType | NftStakeType}):
+  void;
+  unstakeNfts(payload: {ids: string[]; stakeType: StakeType | NftStakeType}):
+  void;
+  unstakeKing(payload: {amount: string;}):
+  Promise<void>;
+  claimKingReward():
+  Promise<void>;
+  stakeUnclaimedRewards(payload: {stakeType: StakeType;}):
+  Promise<void>;
+  claimReward(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<void>;
+  getOwnedLandIdsWithTier():
+  Promise<LandIds[]>;
+  getStakedIds(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<string[]>;
+}
+
+enum CurrentState {
+  connectWallet,
+  stakeLocked,
+  contractFull,
+  amountIsTooBig,
+  waiting,
+  inputIsZero,
+  insufficientBalance,
+  notEnoughFundsInExitPool,
+  notEnoughFundsInExitPoolForFee,
+  ok,
+}
+
+enum RewardClaimState {
+  loading,
+  rewardLocked,
+  ok,
+  noReward,
+}
+
+interface LandIds {
+  id: string,
+  tier: number,
+}
+
+interface StakeData {
+  contractBalance: string;
+  currentRewardEarned: string;
+  ownBalance: string;
+  remainingCapacityForDeposit: BN;
+  remainingCapacityForWithdraw: BN;
+  rewardDistributionTimeLeft: number;
+  rewardMinimumStakeTime: number;
+  stakedBalance: string;
+  unlockTimeLeft: number;
+}
+
+type AllStakeTypes = StakeType | NftStakeType;
+
+export default Vue.extend({
+  components: {
+    Multiselect,
+  },
+  props: {
```

Comment:
If there are required props, please mark them as required, for future use

Reply:
done [6839728](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/6839728c34bb22d46c9a4328c2f8e7680f731db3)